flowmachine/flowmachine/features/subscriber/contact_balance.py

```diff
@@ -81,15 +81,19 @@ def __init__(
         self.subscriber_identifier = subscriber_identifier
         self.exclude_self_calls = exclude_self_calls
 
-        if self.direction in {"both"}:
+        if self.direction == "both":
             column_list = [self.subscriber_identifier, "msisdn_counterpart"]
-            self.tables = tables
+            self.tables = verify_columns_exist_in_all_tables(
```

Comment:
This assignment to `self.tables` (and the verification of the columns) is duplicate, because it also happens in lines 94-95 below. I'd remove this statement here and only set the `column_list` (analogous to the `elif` branch in line 90 below).

Then we can simplify the helper function `verify_columns_exist_in_all_tables` by removing the return statement as well, which isn't needed for the verification that the columns are present.